pkg/staticnetworkconfig/generator.go

```diff
@@ -183,11 +183,21 @@ func (s *StaticNetworkConfigGenerator) validateInterfaceNamesExistence(macInterf
 		if err != nil {
 			return errors.Wrapf(err, "failed to get interface type for interface %s", interfaceName.String())
 		}
-		if lo.Contains([]string{"802-3-ethernet", "ethernet"}, interfaceType.String()) {
+		switch interfaceType.String() {
+		case "802-3-ethernet", "ethernet":
 			if !lo.Contains(interfaceNames, interfaceName.String()) {
 				return errors.Errorf("mac-interface mapping for interface %s is missing", interfaceName.String())
 			}
 			matched = true
+		case "vlan":
```

Comment:
In order for a definition to be associated with a host, a corresponding mac-interface mapping must be found.  Otherwise it is a stale definition.

Reply:
As you answered in the next comment, "With bond, slave connection files are created by nmstate - so they should be matched as ethernet connection files."
AFAIU, vlans should have the same logic. I tried it locally using `nmstatectl gc` and the ethernet interfaces connections were added for both bond slaves and vlan base interface.
Maybe it is a bug that was fixed in a newer nmstate version and we should upgrade the version.
If not, I believe we should document them a bug.

Reply:
This is not the issue for this PR.  We want to validate that the nmstate definition can be associated with a host, and all physical interfaces can be associated with physical host interfaces.  The fact that additional files are created with a bond should not be covered by this PR.

Reply:
The issue is not that additional files are created for a bond, this is the [expected/documented](https://nmstate.io/features/gen_conf.html) behaviour. The issue is the same doesn't happen for a vlan. If the nmstate generator would keep its original behavior and generate a connection file for the vlan undefined parent (as their [documentation](https://nmstate.io/features/gen_conf.html) says) this PR wouldn't be required.
In case the nmstate is up to date I believe it worth at least asking them if this change is intentional or not.
I believe that coupling the code to an undocumented/inconsistent behavior of the nmstate generator makes it fragile.
It of course doesn't block the PR, just a follow up.

Reply:
Nmstate documentation [update](https://nmstate.io/devel/yaml_api.html#vlan-interface). Explaining that starting version 2.2.1 , an un-mentioned vlan parent will not automatically create an interface file.